commcare/src/main/java/org/motechproject/commcare/tasks/builder/DisplayNameHelper.java

```diff
@@ -12,16 +12,33 @@
      * If the name is blank, this will be formatted as "name [config-name]".
      * @param subject the subject with which the display name should start
      * @param name the name of the element (form, case, etc.), if blank will be omitted
+     * @param applicationName the name of the application
      * @param configName the name of the configuration
      * @return the display name
      */
-    public static String buildDisplayName(String subject, String name, String configName) {
+    public static String buildDisplayName(String subject, String name, String applicationName, String configName) {
         if (StringUtils.isBlank(name)) {
             return String.format("%s [%s]", subject, configName);
         } else {
-            return String.format("%s: %s [%s]", subject, name, configName);
+            if (StringUtils.isBlank(applicationName)) {
+                return String.format("%s: %s [%s]", subject, name, configName);
+            } else {
+                return String.format("%s: %s [%s: %s]", subject, name, applicationName, configName);
+            }
         }
     }
+    
+    /**
+     * Builds a display name in the form of "subject: name [config-name]". Example - Received Form: Birth [myConfig].
+     * If the name is blank, this will be formatted as "name [config-name]".
+     * @param subject the subject with which the display name should start
+     * @param name the name of the element (form, case, etc.), if blank will be omitted
+     * @param configName the name of the configuration
+     * @return the display name
+     */
+    public static String buildDisplayName(String subject, String name, String configName) {
```

Comment:
As I've mentioned in my previous comment, the application name should be displayed next to the data source objects (to be consistent with triggers).